src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala

```diff
@@ -140,7 +140,6 @@ class MarathonSchedulerActorTest extends AkkaUnitTest with ImplicitSender with G
         .addTaskDropped(containerName = Some("dropped"))
         .addTaskUnknown(containerName = Some("unknown"))
         .addTaskKilling(containerName = Some("killing"))
-        .addTaskReserved(containerName = Some("reserved"))
```

Comment:
We should keep this test. There are still reserved tasks! We dropped the condition not task with reservations.

Reply:
well this piece of code really works with conditions only so it does not make sense to add residents here IMHO. But in general yes, this PR is definitely missing unit tests coverage. I'll follow up with that once I debug the integration test failures. Feel free to come up with something when I am sleeping :-) I am more than willing to collaborate on this

Reply:
I don't know. I mean we check `assert(nonTerminalTasks.size == 5, "We should have 5 non-terminal tasks")` below. With this in still in place it should be the same, right? In any case I don't understand why you dropped this case.

Reply:
I tend to disagree. This code path is really working only with conditions and verifying condition mapping. You might argue that we should also test task with a long name, task with some label or task with whatever (e.g. reservation). Task does not even have reservation, that's instance. I really believe it does not make sense keeping it here.